File: mobile-wallet-master/core/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/services/SearchService.java
package org.mifos.mobilewallet.core.data.fineract.api.services;

import org.mifos.mobilewallet.core.data.fineract.api.ApiEndPoints;
import org.mifos.mobilewallet.core.data.fineract.entity.SearchedEntity;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Created by naman on 19/8/17.
 */

public interface SearchService {

    @GET(ApiEndPoints.SEARCH)
    Observable<List<SearchedEntity>> searchResources(@Query("query") String query,
            @Query("resource") String resources,
            @Query("exactMatch") Boolean exactMatch);
}

File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/services/SearchService.kt
package org.mifos.mobilewallet.core.data.fineract.api.services

import org.mifos.mobilewallet.core.data.fineract.api.ApiEndPoints
import com.mifos.mobilewallet.model.entity.SearchedEntity
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

/**
 * Created by naman on 19/8/17.
 */
interface SearchService {
    @GET(ApiEndPoints.SEARCH)
    fun searchResources(
        @Query("query") query: String,
        @Query("resource") resources: String,
        @Query("exactMatch") exactMatch: Boolean
    ): Observable<List<SearchedEntity>>
}
